test/modules/http/regressions/http-MaxListenersExceededWarning.test.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @vitest-environment node
+ */
+import { vi, it, expect, beforeAll, afterAll } from 'vitest'
+import http from 'http'
+import { HttpServer } from '@open-draft/test-server/http'
+import { ClientRequestInterceptor } from '../../../../src/interceptors/ClientRequest'
+import {
+  waitForClientRequest,
+} from '../../../helpers'
+
+const httpServer = new HttpServer((app) => {
+  app.get('/', (_req, res) => {
+    let count = 1
+    const i = setInterval(() => {
+      res.write('a')
+      if (++count > 20) {
+        clearInterval(i)
+        res.end()
+      }
+    }, 10)
+  })
+})
+
+const interceptor = new ClientRequestInterceptor()
+
+beforeAll(async () => {
+  interceptor.apply()
+  await httpServer.listen()
+})
+
+afterAll(async () => {
+  interceptor.dispose()
+  await httpServer.close()
+})
+
+it('does ', async () => {
+  const url = httpServer.http.url('/')
+  const warningListener = vi.fn()
+  process.on('warning', warningListener)
```

Comment:
This is an indirect assertion. I think you are onto something with that `connect` event. Can we somehow add a listener to that instead and assert that it's being called once? That would make a nice compliance test.

Reply:
Delightful comment! Fixed.